Graphviz/DOT.sublime-syntax

```diff
@@ -65,56 +394,478 @@ contexts:
-
-  brackets:
+###[ OPERATORS ]###############################################################
+
+  # TODO: graphs contain undirected
+  # TODO: digraphs contain directed
+  # TODO: subgraphs contain assignment
+
+  operators:
+    - match: '--'
+      scope: keyword.operator.edge.graph.dot
+    - match: '->'
+      scope: keyword.operator.edge.digraph.dot
+    - match: '='
+      scope: keyword.operator.assignment.dot
+
+###[ ATTRIBUTE LISTS ]#########################################################
+
+  attributes:
+    - include: fullline-comments-number-sign
+
+    - match: '[,;]'
+      scope: punctuation.separator.sequence.dot
+
+    - include: subgraph-declarations
+    - include: attribute-statements
+    - include: node-statements
+    - include: attribute-lists
+    - include: embedded-html
+    - include: literals
+    - include: operators
+
+  attribute-lists:
     - match: \[
-      scope: punctuation.definition.attributes.begin.dot
-      push:
-        - meta_scope: meta.attributes.dot
-        - match: \]
-          scope: punctuation.definition.attributes.end.dot
-          pop: true
-        - match: '[,;]'
-          scope: punctuation.separator.dot
-        - include: expressions
+      scope: punctuation.section.brackets.begin.dot
+      push: attribute-list-body
+    - include: stray-brackets
 
-  comments:
-    - match: //
-      scope: punctuation.definition.comment.dot
-      push:
-        - meta_scope: comment.line.double-slash.dot
-        - match: $\n?
-          pop: true
-    - match: '#'
-      scope: punctuation.definition.comment.dot
-      push:
-        - meta_scope: comment.line.number-sign.dot
-        - match: $\n?
-          pop: true
-    - match: /\*
-      scope: punctuation.definition.comment.dot
-      push:
-        - meta_scope: comment.block.dot
-        - match: \*/
-          scope: punctuation.definition.comment.dot
-          pop: true
-        - match: ^\s*(\*)(?!/)
-          captures:
-            1: punctuation.definition.comment.dot
+  attribute-list-body:
+    - meta_scope: meta.attributes.dot
+
+    - match: \]
+      scope: punctuation.section.brackets.end.dot
+      pop: 1
+
+    - include: attributes
+
+###[ PROTOTYPES ]##############################################################
+
+  eol-pop:
+    - match: '{{eol}}'
+      pop: 1
+
+###[ ILLEGALS ]################################################################
+
+  illegal-newline:
+    - match: $\n
+      scope: invalid.illegal.unexpected-newline.dot
+      pop: 1
+
+  stray-braces:
+    - match: \}
+      scope: invalid.illegal.stray.dot
+
+  stray-brackets:
+    - match: \]
+      scope: invalid.illegal.stray.dot
+
+###[ VARIABLES ]###############################################################
+
+variables:
+  leading_wspace: (?:^\s*)
+
+  eol: (?:${{line_break}}?)
+  line_break: (?:\n)
+
+  # https://graphviz.org/doc/info/lang.html#ids
+  id: |-
+    (?x:
+      [a-zA-Z_\x{200}-\x{377}][a-zA-Z0-9_\x{200}-\x{377}]*
+    )
```

Comment:
As it is a one-liner pattern, I'd probably do either

```suggestion
  id: |-
    (?x: [a-zA-Z_\x{200}-\x{377}][a-zA-Z0-9_\x{200}-\x{377}]* )
```

or even just

```suggestion
id: (?:[a-zA-Z_\x{200}-\x{377}][a-zA-Z0-9_\x{200}-\x{377}]*)
```

or

```suggestion
id: '[a-zA-Z_\x{200}-\x{377}][a-zA-Z0-9_\x{200}-\x{377}]*'
```

Note: I haven't checked whether any char in `\x200-\x377` is also matched by `\b` potentially causing issues in `id` context.